### DISCUSSION TITLE
iperf3 test tcp have some retry packets in first second

### DISCUSSION THREAD

**Discussion by liaoqian1** [esnet/iperf#1418] [Q&A]
[root@localhost ~]# iperf3 -c 10.168.20.57 -t 10 -i 1 -p 10002 -P 10
Connecting to host 10.168.20.57, port 10002
[  4] local 10.168.20.55 port 57648 connected to 10.168.20.57 port 10002
[  6] local 10.168.20.55 port 57650 connected to 10.168.20.57 port 10002
[  8] local 10.168.20.55 port 57652 connected to 10.168.20.57 port 10002
[ 10] local 10.168.20.55 port 57654 connected to 10.168.20.57 port 10002
[ 12] local 10.168.20.55 port 57656 connected to 10.168.20.57 port 10002
[ 14] local 10.168.20.55 port 57658 connected to 10.168.20.57 port 10002
[ 16] local 10.168.20.55 port 57660 connected to 10.168.20.57 port 10002
[ 18] local 10.168.20.55 port 57662 connected to 10.168.20.57 port 10002
[ 20] local 10.168.20.55 port 57664 connected to 10.168.20.57 port 10002
[ 22] local 10.168.20.55 port 57666 connected to 10.168.20.57 port 10002
[ ID] Interval           Transfer     Bandwidth       Retr  Cwnd
[  4]   0.00-1.00   sec   171 MBytes  1.43 Gbits/sec    0    300 KBytes
[  6]   0.00-1.00   sec   172 MBytes  1.44 Gbits/sec    0    311 KBytes
[  8]   0.00-1.00   sec   171 MBytes  1.44 Gbits/sec    0    310 KBytes
[ 10]   0.00-1.00   sec   171 MBytes  1.44 Gbits/sec    0    281 KBytes
[ 12]   0.00-1.00   sec   171 MBytes  1.43 Gbits/sec    0    301 KBytes
[ 14]   0.00-1.00   sec   175 MBytes  1.46 Gbits/sec    0    462 KBytes
[ 16]   0.00-1.00   sec   174 MBytes  1.46 Gbits/sec    6    288 KBytes
[ 18]   0.00-1.00   sec   173 MBytes  1.45 Gbits/sec    0    397 KBytes
[ 20]   0.00-1.00   sec   175 MBytes  1.46 Gbits/sec    0    454 KBytes
[ 22]   0.00-1.00   sec   174 MBytes  1.46 Gbits/sec    0    436 KBytes
[SUM]   0.00-1.00   sec  1.69 GBytes  14.5 Gbits/sec    6

i want to know the reason for zhe retry.

**Comment by davidBar-On** [comment]
Probably this should have been opened as a [Discussion](https://github.com/esnet/iperf/discussions/), not an Issue.

Regarding the question, to try understanding the reason for the retries you should produce log of the TCP traffic.  Probably on the client side and on the server side and analyze the logs.  Traffic logging can be done using Wireshark, tcpdump, etc.
